src/main/java/org/esbtools/auth/util/EnvironmentUtils.java

```diff
@@ -0,0 +1,99 @@
+package org.esbtools.auth.util;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.naming.NamingException;
+import javax.naming.directory.NoSuchAttributeException;
+import javax.naming.ldap.LdapName;
+import javax.naming.ldap.Rdn;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EnvironmentUtils {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(EnvironmentUtils.class);
+
+    public static final String ENVIRONMENT_SEPARATOR = ",";
+    public static final String LOCATION = "l";
+    public static final String OU = "ou";
+
+    private final String environment;
+    private final String allAccessOu;
+
+    public String getEnvironment() {
+        return environment;
+    }
+
+    public String getAllAccessOu() {
+        return allAccessOu;
+    }
+
+    public EnvironmentUtils(String environment) {
+        this(environment, null);
+    }
+
+    public EnvironmentUtils(String environment, String allAccessOu) {
+        if (environment == null) {
+            throw new NullPointerException("environment cannot be null");
+        }
+
+        this.environment = environment;
+        this.allAccessOu = allAccessOu;
+    }
+
+    public void validateEnvironment(String certificatePrincipal) throws NamingException {
```

Comment:
If we change class name to Environment, this can just be called validate(String...